### PR TITLE
feat(validation): add default flag consistency validation

### LINKED ISSUE #61: Default audio track flag not validated correctly with strictness options
### Description of the bug

When batch processing files with inconsistent default audio track flags, the validation does not respect the configured strictness option. The application uses the last value encountered instead of validating against the strictness option. When the default flag validation is set to 'Warning', no warning banner is displayed.

### Steps to reproduce

1. Add File 1 to batch with Audio track 1 default flag set to false
2. Add File 2 to batch with Audio track 1 default flag set to true  
3. Open Settings and set AudioTrackValidation.DefaultFlag to 'Warning' in validation settings
4. Return to main window and start batch processing
5. Observe no warning banner appears despite the inconsistency

### Actual behavior

No warning banner is displayed. The validation does not trigger regardless of the strictness setting. The last encountered value is used without notifying the user of the inconsistency.

### Expected behavior

When AudioTrackValidation.DefaultFlag is set to:
- **Warning**: A warning banner should be displayed indicating the inconsistency
- **Error**: Batch processing should be blocked
- **Information**: An informational message should be shown
- **Off**: No validation should occur

### Matroska Batch Flow version

Development version

### Application target

Windows App SDK (WinUI 3)

### Distribution method

Build from source (Please specify commit hash in 'Additional information')

### Operating System

Windows 11

### Additional information

Part of tracking issue #60

**Related Code:**
- \src/MatroskaBatchFlow.Core/Models/AppSettings/TrackPropertyValidationSettings.cs\ - Contains \DefaultFlag\ property
- \src/MatroskaBatchFlow.Core/Services/FileValidation/\ - Validation rules implementation

### PR DESCRIPTION
**Pull request by TimGels**
Implement DefaultFlagConsistencyRule to detect inconsistent default flags across media files in a batch. The rule validates that all files have consistent default flag values for each track type (Audio, Video, Subtitle) at corresponding track positions.

The validation respects user-configured severity settings and can be enabled/disabled per track type. When inconsistencies are detected, users receive informative warnings or errors depending on their validation preferences.

Changes:
- Add DefaultFlagConsistencyRule following existing validation pattern
- Register rule in dependency injection container
- Add 15 comprehensive unit tests covering all scenarios
- Validate all track types with configurable severity levels
- Handle edge cases (different track counts, < 2 files)

Resolves #61

### CODE REVIEW

File: src/MatroskaBatchFlow.Core/Services/FileValidation/DefaultFlagConsistencyRule.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                if (referenceFileDefaultFlags[trackIndex] == defaultFlagMatrix[i][trackIndex])
+                    continue;
+
+                yield return new FileValidationResult(
+                    trackSettings.DefaultFlag,
+                    scannedFiles[i].Path,
+                    $"Default flag mismatch in {type} tracks at position {trackIndex + 1}: " +
+                    $"'{referenceFileDefaultFlags[trackIndex]}' (in '{scannedFiles[0].Path}') " +
+                    $"vs '{defaultFlagMatrix[i][trackIndex]}' (in '{scannedFiles[i].Path}')."
+                );
```
[the remark this answers is not in the document]
Comment (TimGels) [reply src/MatroskaBatchFlow.Core/Services/FileValidation/DefaultFlagConsistencyRule.cs]: The warning/Error/info dialog needs to be rethought in the further anyway.

File: src/MatroskaBatchFlow.Core/Services/FileValidation/DefaultFlagConsistencyRule.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        var trackTypes = new[] { TrackType.Audio, TrackType.Video, TrackType.Text };
+
+        foreach (var type in trackTypes)
+        {
+            // Build a matrix of default flags for the specified track type across all scanned files.
+            var defaultFlagMatrix = BuildDefaultFlagMatrix(scannedFiles, type);
+
+            foreach (var result in CompareDefaultFlags(defaultFlagMatrix, scannedFiles, type, settings))
+                yield return result;
+        }
```
[the remark this answers is not in the document]
Comment (TimGels) [reply src/MatroskaBatchFlow.Core/Services/FileValidation/DefaultFlagConsistencyRule.cs]: Good idea!
